pkg/relayer/proxy/server_builder.go

```diff
@@ -42,17 +41,16 @@ func (rp *relayerProxy) BuildProvidedServices(ctx context.Context) error {
 		var serviceEndpoints []relayer.RelayServer
 
 		for _, endpoint := range serviceConfig.Endpoints {
-			url, err := url.Parse(endpoint.Url)
-			if err != nil {
-				return err
-			}
-			supplierEndpointHost := url.Host
+			// TODO(#137): Move `supplierEndpointHost` into a separate config file (could be a part of chains.json if we
+			// should bind different ports for different chains, or just one port as a part of the main config/CLI argument)
+			// `endpoint.Url` should only be used to advertise the endpoint to the internet as a part of stake.
+			supplierEndpointHost := "0.0.0.0:8545"
```

Comment:
@red-0ne We should not do that. I understand this is counterintuitive, but nodes are almost never going to be exposed to the internet directly. Most often, they are going to be used with a reverse proxy - load balancer, nginx, cloudflare, etc. - so there are many options. We should provide a way to change the network interface we bind a port to. Because if we do not, here's what happens:

```
2023/11/27 02:03:35 INFO: starting relay server for service anvil at endpoint http://relayminers:8545
Error: listen tcp 10.96.6.176:8545: bind: cannot assign requested address
2023/11/27 02:03:35 WARN: redundant unsubscribe: observer is closed
```

The node makes a DNS request on the address from the on-chain data, resolves it to an IP address, and tries to bind to that address. Whatever the hostname resolves to should not be used to determine the network interface to bind.

In meanwhile, I add this code back to be able to run e2e tests. We should put it in a config file! I can look into how to do that later.

Reply:
@okdas Makes sense thanks for the explanation. As you stated, a config entry would make sense then. Do you think we should add this in the relay miner config file? Should it be a 1-1 configuration entry w.r.t the hosts exposed on-chain? or just one config entry that handles all of them?

Keep in mind that we probably need one server for each service.